### PR TITLE
Add test for Pkg.build() failure

### PR DESCRIPTION
**Pull request by staticfloat**
Test for https://github.com/JuliaLang/Pkg.jl/pull/1851

### CODE REVIEW

File: test/new.jl [hunk 1]
```diff
@@ -1767,6 +1767,13 @@ end
         @test args == [Pkg.PackageSpec(;name="Foo"), Pkg.PackageSpec(;name="Bar")]
         @test opts == Dict(:verbose => true)
     end
+    
+    # Test package that fails build
+    isolate(loaded_depot=true) do; mktempdir() do tempdir
+        package_path = copy_test_package(tempdir, "FailBuild")
+        Pkg.activate(package_path)
+        @test_throws ErrorException Pkg.build()
```
Comment (KristofferC) [inline test/new.jl:1775]: Maybe this should be a `PkgError` instead.

Comment (staticfloat) [reply test/new.jl:1775]: Isn't the error that gets thrown dependent on what happens inside of `deps/build.jl`?

Comment (KristofferC) [reply test/new.jl:1775]: No, that's a separate process. It will always throw an ErrorException right now (since `error` is used).

Comment (staticfloat) [reply test/new.jl]: Ah, of course.  I thought you were hinting at my `error()` in `build.jl`, so I got confused.

**Pull request merged**